noxfile.py

```diff
@@ -59,6 +59,9 @@ def lint(session):
     session.install("pre-commit")
     session.run("pre-commit", "run", "--all-files")
 
+    # Install current package and dependencies
+    session.install(".")
```

Comment:
Why is this necessary?

Reply:
As a general note: Please include a body to your commit messages, describing why a change is being made.

https://chris.beams.io/posts/git-commit/#why-not-how

Reply:
@pradyunsg To enforce using the correct dependency versions. Installing `nox` -> `pytest` -> `packaging` -> `pyparsing<3`. 
https://github.com/moyiz/packaging/runs/4054317198?check_suite_focus=true

Same goes for `build` package

Reply:
But why does it matter? We're not running packaging during the lint step and even if we did nox would've installed it in a venv separate from nox itself.

Reply:
@layday `nox` indeed creates a virtualenv, but `pyparsing` version seems to be overridden in `python -m pip install build twine` since `packaging` is a dependency of `build` and the latest release forces `pyparsing<3`. This `session.install(".")` is required to enforce the current requirements of `packaging`, rather than latest release.

Reply:
That doesn't answer the why :)

I think I understand what is going on here. Because this project doesn't use a `src`-based tree, the copy of `packaging` from the working directory takes precedence over the one from the nox venv.  So with `packaging` now only supporting `pyparsing >= 3`, `build` crashes.

Instead of pre- or re-installing `packaging` I would suggest using the `build` console script, which would prevent the `cwd` from being added to the Python path:

```diff
diff --git a/noxfile.py b/noxfile.py
index 956201a..1056410 100644
--- a/noxfile.py
+++ b/noxfile.py
@@ -61,7 +61,7 @@ def lint(session):
 
     # Check the distribution
     session.install("build", "twine")
-    session.run("python", "-m", "build")
+    session.run("pyproject-build")
     session.run("twine", "check", *glob.glob("dist/*"))
```

Reply:
@layday Awesome catch, thanks! 
Removed the re-installations in favor of the above console script.